gpflow/optimizers/scipy.py

```diff
@@ -121,12 +126,14 @@ def unpack_tensors(to_tensors: Iterator[tf.Tensor], from_vector: tf.Tensor) -> L
         return values
 
     @staticmethod
-    def assign_tensors(to_tensors: Iterator[tf.Variable], values: Iterator[tf.Tensor]):
+    def assign_tensors(to_tensors: Iterable[tf.Variable], values: Iterable[tf.Tensor]) -> None:
         for tensor, tensor_vector in zip(to_tensors, values):
             tensor.assign(tensor_vector)
 
 
-def _compute_loss_and_gradients(loss_cb: LossClosure, variables: Variables):
+def _compute_loss_and_gradients(
+    loss_cb: LossClosure, variables: Variables
+) -> Tuple[tf.Tensor, tf.Tensor]:
```

Comment:
the return type of [`tf.GradientTape.gradient`](https://www.tensorflow.org/api_docs/python/tf/GradientTape#gradient) is

> a list or nested structure of Tensors (or IndexedSlices, or None), one for each element in sources. Returned structure is the same as the structure of sources.

how do we know which one we're getting?